li-hadoop-plugin/src/main/groovy/com/linkedin/gradle/liazkaban/LiAzkabanDslCompiler.groovy

```diff
@@ -62,26 +63,61 @@ class LiAzkabanDslCompiler extends AzkabanDslCompiler {
       return;
     }
 
-    // since it is a hadoopShell job type, only two properties are relevant.
-    Set<String> relevantProperties = ['type','command'];
+    Map<String,String> azkabanOptions = new HashMap<String,String>();
+    azkabanOptions.put(LiAzkabanJavaProperties.AZKABAN_LINK_WORKFLOW_URL,"\${${LiAzkabanJavaProperties.AZKABAN_LINK_WORKFLOW_URL}}")
+    azkabanOptions.put(LiAzkabanJavaProperties.AZKABAN_LINK_ATTEMPT_URL,"\${${LiAzkabanJavaProperties.AZKABAN_LINK_ATTEMPT_URL}}")
+    azkabanOptions.put(LiAzkabanJavaProperties.AZKABAN_LINK_JOB_URL,"\${${LiAzkabanJavaProperties.AZKABAN_LINK_JOB_URL}}")
+    azkabanOptions.put(LiAzkabanJavaProperties.AZKABAN_LINK_EXECUTION_URL,"\${${LiAzkabanJavaProperties.AZKABAN_LINK_EXECUTION_URL}}")
+    azkabanOptions.put(LiAzkabanJavaProperties.AZKABAN_JOB_INNODES,"\${${LiAzkabanJavaProperties.AZKABAN_JOB_INNODES}}")
+    azkabanOptions.put(LiAzkabanJavaProperties.AZKABAN_JOB_OUTNODES,"\${${LiAzkabanJavaProperties.AZKABAN_JOB_OUTNODES}}")
+
+    // create jvm string of the form -Dkey1=value1 -Dkey2=value2
+    StringBuffer azkabanOpts = new StringBuffer();
+    azkabanOptions.each {
+      key,value -> azkabanOpts.append("-D${key}=${value}");
+        azkabanOpts.append(" ");
```

Comment:
Why not append the space at the end as a part of the previous line, e.g. azkabanOpts.append("-D${key}=${value} "). Then you can also collapse everything down to a single line:
azkabanOptions.each { key, value -> azkabanOpts.append("-D${key}=${value} ") }